tests/system_tests_topology.py

```diff
@@ -535,6 +535,8 @@ def on_message(self, event):
                 router = 'C'
             elif event.receiver == self.routers['D']['mgmt_receiver']:
                 router = 'D'
+            else:
+                raise Exception("Unexpected event receiver")
```

Comment:
`router` may be used before initialization

From the actual context this doesn't appear possible, but yeah there is a hole in the code where router is never set.  Rather than suppress the warning I added the Exception as a "defensive coding" approach.  This test has a history of being somewhat flaky so I'm sure we'll be hacking at it at some point in the future. Should we change the code in some way where we add a new event receiver then the existing if statement would fail silently and `router` would in fact be uninitialized.  The exception would prevent that foot-gun.